cmd/remote_client.go

```diff
@@ -262,21 +264,95 @@ func (cli *Client) RemoteLogin(c *clipkg.Context) error {
 	return cli.errorOut(err)
 }
 
+func extractContractAddressFromAssignmentSyntax(flag string) (string, error) {
+	equalSplit := strings.Split(flag, "=")
+	if len(equalSplit) != 2 {
+		return "", errors.New("too many '=' in" + flag)
+	}
+	return equalSplit[1], nil
+}
+
+// extractEthereumAddressFromWithdrawalCommandline takes the arg containing the
+// specific-contract-address flag, and any following args, and returns the
+// ethereum address as a string, if present, or the empty string if not.
+// Assumes the contract-address flag comes at the end...
+func extractEthereumAddressFromWithdrawalCommandline(args []string) (string, error) {
+	flag := args[0]
+	if strings.Contains(flag, "=") {
+		if len(args) > 1 {
+			return "", fmt.Errorf("too many args for %s", flag)
+		}
+		return extractContractAddressFromAssignmentSyntax(flag)
+	}
+	if len(args) != 2 {
+		return "", fmt.Errorf("wrong number of args for %s", flag)
+	}
+	return args[1], nil
+}
+
+// ParseWithdrawalContractAddress is exposed only for testing purposes. Do not
+// use externally. Checks for valid specification of correctly-capitalized
+// contract address. If it finds one, it returns that. Otherwise, it returns the
+// default ("empty") Address, and any errors it encountered in the flag or
+// during parsing.
+//
+// TODO: go's flags library has a lot of useful parsing tools. Can we use that?
```

Comment:
The CLI library that we use provides some nice flag parsing functionality, so that we don't have to build it ourselves: https://github.com/smartcontractkit/chainlink/blob/features/txmanager-round-robin-addresses/cmd/remote_client.go#L333

Reply:
Thanks, I have simplified that.